pkg/cluster/k8sres.go

```diff
@@ -893,10 +894,20 @@ func (c *Cluster) getPodEnvironmentSecretVariables() ([]v1.EnvVar, error) {
 		return secretPodEnvVarsList, nil
 	}
 
-	secret, err := c.KubeClient.Secrets(c.Namespace).Get(
-		context.TODO(),
-		c.OpConfig.PodEnvironmentSecret,
-		metav1.GetOptions{})
+	secret := &v1.Secret{}
+	err := retryutil.Retry(c.OpConfig.ResourceCheckInterval, c.OpConfig.ResourceCheckTimeout,
+		func() (bool, error) {
+			var err error
+			secret, err = c.KubeClient.Secrets(c.Namespace).Get(
+				context.TODO(),
+				c.OpConfig.PodEnvironmentSecret,
+				metav1.GetOptions{})
+			if err != nil {
+				return false, err
```

Comment:
`err` shouldn't be returned here, otherwise `Retry` will not work and existed immediately.
Probably only `apierrors.IsNotFound` could be ignored only

Reply:
Excellent observation. I changed it to check for the is not found error, and to wrap it when needed. I also added a test.